airbyte_cdk/sources/streams/http/exceptions.py

```diff
@@ -59,3 +59,24 @@ class DefaultBackoffException(BaseBackoffException):
 
 class RateLimitBackoffException(BaseBackoffException):
     pass
+
+
+class DNSResolutionError(BaseBackoffException):
```

Comment:
I don't see where this exception class is raised. Can you help me understand this?

Does our code need to raise this exception in order for its handling to trigger? Or is it going to be picked up automatically somehow?